shared/js/media/video_player.js

```diff
@@ -105,19 +105,25 @@ function VideoPlayer(container) {
   var orientation = 0; // current player orientation
 
   // These are the raw (unrotated) size of the poster image, which
-  // must have the same size as the video.
+  // may have the same size as the video. We may find when we actually
+  // load the video they are different and resize the video element.
   var videowidth, videoheight;
+  var posterwidth, posterheight;
 
   var playbackTime;
   var capturedFrame;
 
-  this.load = function(video, posterimage, width, height, rotate, timestamp) {
+  this.load = function(video, posterimage, width, height, rotate,
+                       posterWidth, posterHeight, timestamp)
```

Comment:
Hmm, perhaps the best thing to do is to remove it. It has limited use considering the caller won't have preloaded the video for its metadata. The only use case one would have for it is for camera which knows the size in advance, but 1) if there isn't extra flickering in gallery, camera doesn't need it, and 2) if there is extra flickering in gallery, we will want same sized posters.